File: hj/week4/BinarySearch .cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;

        while (left <= right) {
            int mid = (left + right) / 2;

            if (target > nums[mid]) {
                left = mid + 1;
            }
            else if (target < nums[mid]) {
                right = mid - 1;
            }
            else
                return mid;
        }

        return -1;
    }
};

File: 2025/Jun/binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            curr = nums[mid]
            if curr == target:
                return mid
            if target < curr:
                right = mid - 1
            else:
                left = mid + 1
        return -1
